thirdparty/rocksdb/all/patches/arm7.patch

```diff
@@ -4,7 +4,7 @@ index 225e3fa72..cd5f935f1 100644
 +++ b/utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h
 @@ -131,6 +131,10 @@ static inline tokutime_t toku_time_now(void) {
    uint64_t result;
-   __asm __volatile__("mrs %[rt], cntvct_el0" : [ rt ] "=r"(result));
+   __asm __volatile__("mrs %[rt], cntvct_el0" : [rt] "=r"(result));
```

Comment:
The main reason I updated the patch here was because when I tried to create the RocksDB conan package, it complained about "expected" vs "actual" with line 7. Here was the error:

~~~bash
rocksdb/8.10.2@minifi/develop: WARN: /home/ubuntu/.conan2/p/rocksb6dac60094a95/s/patches/arm7.patch: file 1/1:	 b'utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h'
rocksdb/8.10.2@minifi/develop: WARN: /home/ubuntu/.conan2/p/rocksb6dac60094a95/s/patches/arm7.patch:  hunk no.1 doesn't match source file at line 132
rocksdb/8.10.2@minifi/develop: WARN: /home/ubuntu/.conan2/p/rocksb6dac60094a95/s/patches/arm7.patch:   expected: b'  __asm __volatile__("mrs %[rt], cntvct_el0" : [ rt ] "=r"(result));'
rocksdb/8.10.2@minifi/develop: WARN: /home/ubuntu/.conan2/p/rocksb6dac60094a95/s/patches/arm7.patch:   actual  : b'  __asm __volatile__("mrs %[rt], cntvct_el0" : [rt] "=r"(result));'
rocksdb/8.10.2@minifi/develop: /home/ubuntu/.conan2/p/rocksb6dac60094a95/s/patches/arm7.patch: source file is different - b'utilities/transactions/lock/range/range_tree/lib/portability/toku_time.h'
rocksdb/8.10.2@minifi/develop: ERROR: 
Package 'e9ee71a0ada9817d6bde6ac5f5434fa0e06b4166' build failed
rocksdb/8.10.2@minifi/develop: WARN: Build folder /home/ubuntu/.conan2/p/b/rocks7495fe807ea4d/b/build/Release
*********************************************************
Recipe 'rocksdb/8.10.2@minifi/develop' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter, report it at https://github.com/conan-io/conan-center-index/issues
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: rocksdb/8.10.2@minifi/develop: Error in build() method, line 177
	apply_conandata_patches(self)
	ConanException: Failed to apply patch: /home/ubuntu/.conan2/p/rocksb6dac60094a95/s/patches/arm7.patch
~~~

By making that slight adjustment to the patch, when I created the custom RocksDB conan package, it was able to apply this patch successfully while creating that RocksDB conan package.

Reply:
I also do want to point out that when we build MiNiFi from source using the standalone CMake approach and we build that RocksDB applying its patches with this arm7.patch slightly updated, the BundledRocksDB external lib is built successfully too.

Reply:
Heres the result from creating the custom RocksDB conan package after applying this patch and the other relevant patches:

~~~bash
~/src/open_source/nifi-minifi-cpp/thirdparty/rocksdb/all (MINIFICPP-2346-P1)$ conan create . --user=minifi --channel=develop --version=8.10.2 -pr=../../../etc/conan/profiles/release-linux

# Output (Showing RocksDB conan package created successfully)
rocksdb/8.10.2@minifi/develop: package(): Packaged 104 '.h' files
rocksdb/8.10.2@minifi/develop: package(): Packaged 1 '.a' file: librocksdb.a
rocksdb/8.10.2@minifi/develop: package(): Packaged 1 file: COPYING
rocksdb/8.10.2@minifi/develop: package(): Packaged 1 '.leveldb' file: LICENSE.leveldb
rocksdb/8.10.2@minifi/develop: package(): Packaged 1 '.Apache' file: LICENSE.Apache
rocksdb/8.10.2@minifi/develop: Created package revision 7b0eeda125026e144c7d154e3bffbaa9
rocksdb/8.10.2@minifi/develop: Package 'e9ee71a0ada9817d6bde6ac5f5434fa0e06b4166' created
rocksdb/8.10.2@minifi/develop: Full package reference: rocksdb/8.10.2@minifi/develop#d31518436210c3f016be820242c23550:e9ee71a0ada9817d6bde6ac5f5434fa0e06b4166#7b0eeda125026e144c7d154e3bffbaa9
rocksdb/8.10.2@minifi/develop: Package folder /home/ubuntu/.conan2/p/b/rocksaebf4a8af630b/p
WARN: deprecated: Usage of deprecated Conan 1.X features that will be removed in Conan 2.X:
WARN: deprecated:     'cpp_info.names' used in: bzip2/1.0.8, zlib/1.3.1, lz4/1.9.4, rocksdb/8.10.2@minifi/develop, zstd/1.5.2
WARN: deprecated:     'cpp_info.build_modules' used in: lz4/1.9.4, bzip2/1.0.8
WARN: deprecated:     'env_info' used in: zstd/1.5.2, bzip2/1.0.8
~~~

Reply:
@szaszm I just learned that Conan supports fuzzy patching, so even if there are minor differences between patch code and actual source code, we can still apply the patch. Therefore, we can keep the original **arm7.patch** based on MiNiFi **main** branch and apply it to our rocksdb conan package. I reverted the **arm7.patch** back to how it is in **main** branch. I was able to create a custom **rocksdb** conan package and not run into the error I mentioned earlier.

I will double check that I can build MiNiFi using conan build and then conan create. I will also check I can build MiNiFi using standalone CMake. I will run the CTESTs too. Then I will push the latest code updates.

Reply:
@szaszm in my latest commit "[374fff2](https://github.com/apache/nifi-minifi-cpp/pull/1793/commits/374fff2415d5a1099efb95a0ec198d35472259ea)", I reverted **arm7.patch** back to how it was in **main** branch.